### PR TITLE
[SQL] Set sessionState in QueryExecution.

### PR DESCRIPTION
**Pull request by yhuai**
This PR sets the SessionState in HiveContext's QueryExecution. So, we can make sure that SessionState.get can return the SessionState every time.

### CODE REVIEW

**Comment by yhuai** [comment]
test this please

**Comment by yhuai** [comment]
test this please

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala [hunk 1]
```diff
@@ -27,6 +27,7 @@ import org.apache.hadoop.hive.metastore.{Warehouse, TableType}
 import org.apache.hadoop.hive.metastore.api.{Table => TTable, Partition => TPartition, FieldSchema}
 import org.apache.hadoop.hive.ql.metadata._
 import org.apache.hadoop.hive.ql.plan.CreateTableDesc
+import org.apache.hadoop.hive.ql.session.SessionState
```
Comment (chenghao-intel) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala]: Keep it unchange?

**Comment by chenghao-intel** [comment]
The change looks reasonable to me, except an unnecessary import

**Comment by marmbrus** [comment]
Thanks!  Merged to master and 1.3

**Pull request closed without merging**